### ISSUE TITLE
[feat] interactive input when `init --apply`

### ISSUE THREAD

**Issue by loynoir** [twpayne/chezmoi#2235]
## Is your feature request related to a problem? Please describe.

```sh
$ chezmoi init --apply dotfile-repo
chezmoi: template: dot_XXX.tmpl:XXX:XXX: executing "dot_XXX.tmpl" 
at <.XXX.XXX>: map has no entry for key "XXX"
```

## Describe the solution you'd like
Provide interactive input.

Dotfiles management is very like project kickstarter.

Project kickstarter [yeoman](https://www.npmjs.com/package/yo) provides beautiful interactive input using [inquirer](https://www.npmjs.com/package/inquirer). Tab-able, and input-able, default values, etc.

I think it would be awesome to got that beautiful interactive input in chezmoi.

**Comment by BasixKOR** [comment]
There's prompt functions for templates such as this: https://www.chezmoi.io/reference/templates/init-functions/promptString/

**Comment by twpayne** [comment]
As @BasixKOR says, chezmoi already provides interactive input with `init --apply` with the `prompt*` template functions. These are very basic, but this is by design:

* chezmoi tries to run everywhere, including places without a rich terminal or a Unicode font.
* Creating a config file should be a rare event: a maximum of once per new machine.

Given this, I don't think it's worth adding rich terminal input functions to chezmoi.

**Comment by twpayne** [comment]
Fixed with #2359.
